migration-engine/connectors/sql-migration-connector/src/flavour.rs

```diff
@@ -0,0 +1,253 @@
+//! SQL flavours implement behaviour specific to a given SQL implementation (PostgreSQL, SQLite...),
+//! in order to avoid cluttering the connector with conditionals. This is a private implementation
+//! detail of the SQL connector.
+
+use crate::{
+    catch, connect, database_info::DatabaseInfo, CheckDatabaseInfoResult, SqlError, SqlResult, SystemDatabase,
+};
+use futures::future::TryFutureExt;
+use migration_connector::{ConnectorError, ConnectorResult};
+use once_cell::sync::Lazy;
+use quaint::{
+    connector::{ConnectionInfo, MysqlUrl, PostgresUrl, Queryable},
+    single::Quaint,
+};
+use regex::RegexSet;
+use sql_schema_describer::{SqlSchema, SqlSchemaDescriberBackend};
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use url::Url;
+
+pub(crate) fn from_connection_info(connection_info: &ConnectionInfo) -> Box<dyn SqlFlavour + Send + Sync + 'static> {
```

Comment:
Could this be part of the trait?

Reply:
I think that would make it non-object safe? I can check

Reply:
Yep, confirmed.